src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1632,10 +1635,20 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
 
     if (create.select && create.isView())
     {
-        // Expand CTE before filling default database
-        ApplyWithSubqueryVisitor(getContext()).visit(*create.select);
-        AddDefaultDatabaseVisitor visitor(getContext(), current_database);
-        visitor.visit(*create.select);
+        if (getContext()->getSettingsRef()[Setting::allow_experimental_analyzer])
+        {
+            InterpreterSelectQueryAnalyzer interpreter(create.select->clone(), getContext(), SelectQueryOptions{}.analyze());
+
+            /// Rewrite SELECT query to apply CTEs and expressions from WITH and qualify all the table expressions
+            create.set(create.select, interpreter.getQueryTree()->toAST());
```

Comment:
This line also leads to the `*` matcher expansion to the column list. Is it okay?

Reply:
also I removed tables/joins aliasing, however it may create compatibility issues since table definition for VIEW now different - with resolved projection - they are fully qualified now...